From: SeongJae Park <sj@kernel.org>
Date: Wed, 28 Jan 2026 17:07:48 -0800
Subject: [PATCH] mm/damon/core: align region size uppr limit on min_region_sz

Otherwise, regions of size not same to min_regions_sz could be made.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 29f81a1bb0d43..2866366085c94 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1530,8 +1530,9 @@ static unsigned long damon_region_sz_limit(struct damon_ctx *ctx)
 
 	if (ctx->attrs.min_nr_regions)
 		sz /= ctx->attrs.min_nr_regions;
-	if (sz < ctx->min_region_sz)
+	if (!sz)
 		sz = ctx->min_region_sz;
+	sz = ALIGN(sz, ctx->min_region_sz);
 
 	return sz;
 }
-- 
2.47.3

